JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/CompileAndRunBean.java

```diff
@@ -123,9 +125,12 @@ public String run(String projectHash) {
                     mes += "Stderr: " + help + "\n";
                 }
                 java.io.File projectFile = new java.io.File(projectHash);
+                process.waitFor();
```

Comment:
The solution is still not complete.

Look: you initiated the process, read the output once, than hanged waiting for the process to finish.
You are not reading while the process is running.